src/psyclone/domain/lfric/transformations/lfric_alg_invoke_2_psy_call_trans.py

```diff
@@ -58,16 +67,323 @@ def validate(self, node, options=None):
+                "LFRicKernelContainer as the root or first child of the "
+                "root but this was not found.")
+        return kernel_metadata
+
+    @staticmethod
+    def _add_arg(arg, arguments):
+        '''Utility method to add argument arg to the arguments list as long as
+        it conforms to the expected constraints.
+
+        :param arg: the argument that might be added to the arguments \
+            list.
+        :type arg: :py:class:`psyclone.psyir.nodes.Reference`
+        :param arguments: the arguments list that the argument might \
+            be added to.
+        :type arguments: List[:py:class:`psyclone.psyir.nodes.Reference`]
+
+        :raises InternalError: if the arg argument is an unexpected \
+            type.
+
+        '''
+        sym_maths = SymbolicMaths.get()
+
+        if isinstance(arg, Literal):
+            # Literals are not passed by argument.
+            pass
+        elif isinstance(arg, (Reference, ArrayReference)):
+            for existing_arg in arguments:
+                if sym_maths.equal(arg, existing_arg):
+                    break
+            else:
+                arguments.append(arg.copy())
+        elif isinstance(arg, CodeBlock):
+            arguments.append(arg.copy())
+        else:
+            raise InternalError(
+                f"Expected Algorithm-layer kernel arguments to be "
+                f"a literal, reference or array reference, but "
+                f"found '{type(arg).__name__}'.")
+
+    def get_arguments(self, node, options=None, check_args=False):
+        '''Creates the LFRic processed (lowered) argument list from the
+        argument lists of the kernel functors within the invoke call
+        and the kernel metadata.
+
+        :param node: an LFRic algorithm invoke call.
+        :type node: :py:class:`psyclone.domain.lfric.algorithm.psyir.\
+            LFRicAlgorithmInvokeCall`
+        :param options: a dictionary with options for transformations.
+        :type options: Optional[Dict[str, Any]]
+        :param options["kernels"]: this option provides a list of \
+            LFRic kernels for this LFRic Algorithm Invoke call.
+        :type options["kernels"]: \
+            List[:py:class:`psyclone.psyir.nodes.Container`]
+        :param bool check_args: if True, checks the number of kernel \
+            functor arguments matches the number expected by the kernel \
+            metadata. Defaults to False.
+
+        :raises GenerationError: if the number of arguments in the \
+            invoke does not match the expected number of arguments \
+            specified by the metadata.
+
+        '''
+        # TODO #1618. This routine could/should make use of the new
+        # LFRicArgOrder class if and when it makes its way onto trunk.
+
+        const = LFRicConstants()
+        kernels = options["kernels"]
+
+        arguments = []
+        quad_arguments = []
+        stencil_arguments1 = []
```

Comment:
Please add comments describing the purpose of these vars. Also, can `stencil_arguments{1,2}` have more helpful names?

Reply:
OK. I've added comments and renamed the stencil* vars.